Assets/Mirror/Runtime/Transport/Websocket/WebsocketTransport.cs

```diff
@@ -44,18 +48,34 @@ public void Awake()
         public override bool ClientConnected() { return client.IsConnected; }
         public override void ClientConnect(string host)
         {
-            client.Connect(new Uri($"ws://{host}:{port}"));
-        }
-        public override bool ClientSend(int channelId, byte[] data)
-        {
-            client.Send(data); return true;
+            if (Secure)
+            {
+                client.Connect(new Uri($"wss://{host}:{port}"));
+            }
+            else
+            {
+                client.Connect(new Uri($"ws://{host}:{port}"));
+            }
         }
+        public override bool ClientSend(int channelId, byte[] data) { client.Send(data); return true; }
         public override void ClientDisconnect() { client.Disconnect(); }
 
         // server
         public override bool ServerActive() { return server.Active; }
         public override void ServerStart()
         {
+            
+            if (Secure)
+            {
+                server._secure = Secure;
+                server._sslConfig = new Server.SslConfiguration
+                {
+                    Certificate = new System.Security.Cryptography.X509Certificates.X509Certificate2(Application.dataPath + CertificatePath, CertificatePassword),
```

Comment:
instead of loading the certificate ourselfs,  make the X509Certificate2  a property of this object, and let the user load the certificate in any way they want.